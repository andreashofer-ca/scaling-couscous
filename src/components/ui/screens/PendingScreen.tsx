import { BusyIndicator } from '@ui5/webcomponents-react'
import { FC } from 'react'
import BusyIndicatorSize from '@ui5/webcomponents/dist/types/BusyIndicatorSize.js'
import { PendingScreenProps } from 'interfaces/properties'
import { BUSY_INDICATOR_DELAY } from 'shared/constants'

const PendingScreen: FC<PendingScreenProps> = ({ title, description }) => (
  <BusyIndicator
    delay={BUSY_INDICATOR_DELAY}
    active
    text={description}
    title={title}
    size={BusyIndicatorSize.L}
    className="full-width-busy-indicator"
  />
)

export default PendingScreen
