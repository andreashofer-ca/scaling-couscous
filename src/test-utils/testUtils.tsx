import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
import { render, RenderOptions } from '@testing-library/react'
import { ThemeProvider } from '@ui5/webcomponents-react'
import React, { ReactElement } from 'react'
import { BrowserRouter } from 'react-router-dom'
import { ConfigContext } from 'hooks/config/useConfig'
import config from 'test/config'

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
      },
    },
  })
  return (
    <QueryClientProvider client={queryClient}>
      <ThemeProvider>
        <ConfigContext.Provider value={config}>
          <BrowserRouter>{children}</BrowserRouter>
        </ConfigContext.Provider>
      </ThemeProvider>
    </QueryClientProvider>
  )
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: AllTheProviders, ...options })

const AllTheProvidersWithoutBrowserRouter = ({ children }: { children: React.ReactNode }) => {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
      },
    },
  })
  return (
    <QueryClientProvider client={queryClient}>
      <ThemeProvider>
        <ConfigContext.Provider value={config}>{children}</ConfigContext.Provider>
      </ThemeProvider>
    </QueryClientProvider>
  )
}

const customRenderWithoutBrowserRouter = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>,
) => render(ui, { wrapper: AllTheProvidersWithoutBrowserRouter, ...options })

export * from '@testing-library/react'
export { customRender as render }

export { customRenderWithoutBrowserRouter as renderWithoutRouter }
