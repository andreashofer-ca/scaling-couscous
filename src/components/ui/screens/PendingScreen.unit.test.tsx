import { describe, it, expect } from 'vitest'
import PendingScreen from 'components/ui/screens/PendingScreen'
import { render } from 'test-utils/testUtils'

const title = 'Title'
const description = 'A description'

describe('Pending Screen', () => {
  const { container } = render(<PendingScreen title={title} description={description} />)

  it('should match the snapshot', () => {
    expect(container).toMatchSnapshot()
  })
})
