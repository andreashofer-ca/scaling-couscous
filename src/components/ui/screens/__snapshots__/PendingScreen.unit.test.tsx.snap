// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Pending Screen > should match the snapshot 1`] = `
<div>
  <ui5-busy-indicator
    active=""
    class="full-width-busy-indicator"
    delay="500"
    desktop=""
    size="L"
    text="A description"
    text-placement="Bottom"
    title="Title"
    ui5-busy-indicator=""
  />
</div>
`;
